clarity-ext-scripts/clarity_ext_scripts/covid/rtpcr_analysis_service.py

```diff
@@ -26,82 +33,121 @@ class NegativeControlWasPositive(AnalysisServiceException):
     pass
 
 
+class FailedControl(AnalysisServiceException):
+    pass
+
+
 class RTPCRAnalysisService(object):
+    """
+    Create a new RTPCRAnalysis Service:
+        RTPCRAnalysisService(covid_reporter_key="FAM",
+                             internal_control_reporter_key="HEX")
+    """
 
-    HEX_THRESHOLD = 32
-    FAM_THRESHOLD = 38
+    INTERNAL_CONTROL_THRESHOLD = 32
+    COVID_CONTROL_THRESHOLD = 38
 
-    def __init__(self):
-        pass
+    def __init__(self, covid_reporter_key, internal_control_reporter_key):
+        self._covid_reporter_key = covid_reporter_key
+        self._internal_control_reporter = internal_control_reporter_key
```

Comment:
The two names are now inconsistent (only the argument ends with "key")).

Reply:
Btw. when I asked about calling it "key" I was just trying to understand the "reporter" name, that you meant that it was a "property" or "key". So I imagined the names could be "covid_key" and "internal_control_key".

Reply:
I'll fix the inconsistency. The reason I put reporter in the name is to make it clearer that it is a "reporter color", from what I understand FAM and HEX are actually dyes, used to detect the signal. I think it is good to keep it there, since it is not actual a "covid key value pair", but rather the readout of a color indicating the amount covid-19 RNA in the sample.